src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -369,7 +400,7 @@ public unsafe virtual OperationStatus EncodeUtf8(ReadOnlySpan<byte> utf8Source,
                         // Check Ascii cache.
                         byte[] encodedBytes = GetAsciiEncoding((byte)nextScalarValue);
 
-                        if (ReferenceEquals(encodedBytes, s_noEscape))
+                        if (ReferenceEquals(encodedBytes, Array.Empty<byte>()))
```

Comment:
Is `Array.Empty<T>()` okay for reference equality? I noticed https://github.com/dotnet/corefx/pull/39708/files added the suppression of the CA1825 warning to TextEncoder for an empty array allocation.

Reply:
I verified it worked locally. Originally I wasn't sure so I created a singleton. Now I'm having second thoughts :)

I'll revert this change back to what I had previously and add additional comments.